lib/safira/contest/contest.ex

```diff
@@ -145,22 +145,34 @@ defmodule Safira.Contest do
   end
 
   def list_leaderboard do
-    Repo.all(from a in Safira.Accounts.Attendee, 
+    Repo.all(from a in Safira.Accounts.Attendee,
       where: not (is_nil a.user_id))
     #|> Repo.preload(:badges)
     |> Repo.preload([badges: from(b in Badge, where: b.type != ^0)])
     |> Enum.map(fn x -> Map.put(x, :badge_count, length(Enum.filter(x.badges,fn x -> x.type != 0 end))) end)
     |> Enum.sort(&(&1.badge_count >= &2.badge_count))
   end
 
+  def list_daily_leaderboard(date) do
+    Repo.all(
+      from a in Safira.Accounts.Attendee,
+        join: r in Redeem, on: a.id == r.attendee_id,
+        join: b in Badge, on: r.badge_id == b.id,
+        where: not(is_nil a.user_id) and fragment("?::date", r.inserted_at) == ^date and b.type != ^0,
+        preload: [badges: b]
```

Comment:
Just a quick question from someone who's not 100% literate in elixir. I understand the pin operator is used exactly to "pin" variables to a value, however why is it being used here in logical statements such as ` == ^date`? Also, why is it used with a constant, such as happens with the 0 on `b.type != ^0`.

Reply:
With `^`, you can inject external values and expressions into a query. That's the `^date` case.
As for the `^0`, I confess that it doesn't seem necessary here, but I used it since in functions of the same kind, in the same file, it is also used.